gadgets/src/batched_is_zero.rs

```diff
@@ -26,6 +27,29 @@ pub struct BatchedIsZeroConfig {
     pub nonempty_witness: Column<Advice>,
 }
 
+impl BatchedIsZeroConfig {
+    /// Returns the list of ALL the gadget advice columns.
+    fn advice_columns(&self) -> Vec<Column<Advice>> {
+        vec![self.is_zero, self.nonempty_witness]
+    }
+
+    /// Returns the String annotations associated to each column of the gadget.
+    pub fn annotations(&self) -> Vec<String> {
+        vec![
+            String::from("GADGETS_BATCHED_IS_ZERO_is_zero"),
+            String::from("GADGETS_BATCHED_IS_ZERO_nonempty_witness"),
+        ]
+    }
+
+    /// Annotates columns of this gadget embedded within a circuit region.
+    pub fn annotate_columns_in_region<F: Field>(&self, region: &mut Region<F>, prefix: &str) {
+        self.advice_columns()
+            .iter()
+            .zip(self.annotations().iter())
+            .for_each(|(&col, ann)| region.name_column(|| format!("{}_{}", prefix, ann), col))
+    }
+}
```

Comment:
This looks a bit odd. Looks like you want to implement the `LookupTable` trait for the `Config`. Is that right??
I see happens the same for the rest of the configs. 

This is definitely an usage of the LookupTable trait that I did not imagine when I coded it LOL! 
I'm not sure which is the best way to proceed. Since you have a good abstraction but is only used once anyway. 

WDYT @ed255 ?

Reply:
Why do you think it looks odd?  I actually suggested Kimi to follow this approach :stuck_out_tongue_closed_eyes: which is the same that Leo did here https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1167/files and I think it looks very clean.
Although the `LookupTable` trait shares these 3 methods, the methods themselves have nothing specific about `LookupTable`; but are actually related to "CollectionOfColumns".  I think it's good that we have these methods in `LookupTable` because each lookup table is a "CollectionOfColumns" and it gives us the convenience to annotate the tables uniformly.  This is the same approach but for gadgets, with a small change: a gadget is in a circuit, so it's necessary to add a prefix to distinguish multiple instances of the same gadget.

I don't follow when you say "you have a good abstraction but is only used once anyway", as the same pattern is applied to all the Config types, so this abstraction is used many times.  Maybe you mean that the methods `advice_columns` and `annotations` are only called once? (in `annotate_columns_in_region`).  If that's what you mean, then I agree, maybe having a single function instead of three would be enough, and the two vectors could be embedded into the `annotate_columns_in_region` like this:
```rust
    /// Annotates columns of this gadget embedded within a circuit region.
    pub fn annotate_columns_in_region<F: Field>(&self, region: &mut Region<F>, prefix: &str) {
        [
            ([self.is_zero,"GADGETS_BATCHED_IS_ZERO_is_zero"),
            (self.nonempty_witness,"GADGETS_BATCHED_IS_ZERO_nonempty_witness"),
        ]
            .iter()
            .zip(self.annotations().iter())
            .for_each(|(&col, ann)| region.name_column(|| format!("{}_{}", prefix, ann), col))
    }
```

If this is what you mean, then I agree that this would make the code look cleaner.

Reply:
fixed in [468566c](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1152/commits/468566cf93f158637ece59266654a4739e09f521)

Reply:
> Maybe you mean that the methods advice_columns and annotations are only called once? (in annotate_columns_in_region). If that's what you mean, then I agree, maybe having a single function instead of three would be enough, and the two vectors could be embedded into the annotate_columns_in_region like this:

Exactly!!!